agent/src/main.cpp

```diff
@@ -104,6 +104,13 @@ int main(int argc, char **argv)
     }
 
     SoftwareContainerAgent agent(mainContext, preloadCount, shutdownContainers, timeout);
+    if (!agent.checkWorkspace()) {
```

Comment:
The agent has a reference to a workspace and is also responsible for providing it to containers. With this check, the interface of the workspace is exposed to the user of the agent, which I don't think is good. I suggest we move checks for the workspace to the agent and have it self be responsible for checking its own pre-requisites. I.e the user of the agent really wants to do something else than worry about status of members of the agent, the check can be done when that something else is done by the user instead.